src/Common/tests/System/Net/VirtualNetwork/VirtualNetwork.cs

```diff
@@ -31,14 +34,15 @@ public void ReadFrame(bool server, out byte[] buffer)
                 packetQueue = _serverWriteQueue;
             }
 
-            semaphore.Wait(TestConfiguration.PassingTestTimeoutMilliseconds);
-            buffer = packetQueue.Dequeue();
+            semaphore.Wait(WaitForReadDataTimeoutMilliseconds);
+
+            Debug.Assert(packetQueue.TryDequeue(out buffer), "Packet queue: TryDequeue failed.");
```

Comment:
```C#
Debug.Assert(packetQueue.TryDequeue(out buffer), "Packet queue: TryDequeue failed.");
```
needs to be:
```C#
bool dequeued = packetQueue.TryDequeue(out buffer);
Debug.Assert(dequeued, "Packet queue: TryDequeue failed");
```
Otherwise, the call to TryDequeue gets removed in release builds.